5.5/examples/replica/README.md

```diff
@@ -26,8 +26,12 @@ This template will start with one MySQL master server and one slave server.
 ## Persistent storage
 
 In order to provide the persistent storage for the MySQL master server, the administrator
-of OpenShift needs to create a PersistentVolume that you can claim. This example requires a PersistentVolume of size 512m be available.
-To learn more about how to create PersistentVolume, refer to [OpenShift documentation](https://docs.openshift.org/latest/install_config/persistent_storage/persistent_storage_nfs.html)
+of OpenShift needs to create a PersistentVolume that you can claim. This example requires two persistent volumes of size 512m be available.
+To learn more about how to create PersistentVolume, refer to the [OpenShift documentation](https://docs.openshift.org/latest/install_config/persistent_storage/persistent_storage_nfs.html).
+
+The slave use persistent storage to serve as a hot backup and to optimize
+restart performance, such that only the data written while the slave was down
+needs to be replicated and not the entire database.
```

Comment:
> One doesn't need to change the template do use emptyDir, because the template defines a PVC, not a PV.

I don't understand what your statement here...  PVC and EmptyDir are mutually exclusive.  To use an emptydir instead of a PV, you need to remove the PVC and add an EmptyDir:

https://github.com/openshift/origin/blob/master/examples/db-templates/mongodb-persistent-template.json#L164-L170

vs

https://github.com/openshift/origin/blob/master/examples/db-templates/mongodb-ephemeral-template.json#L147-L153

so you either need to edit the template, or edit the DC after creating it.  Editing the template would be preferred.  You don't define an "emptydir PV" and then use a PVC to claim it.

> In real use scenarios, if you already have data on the master, replication won't work without manual intervention. See the documentation on how to enable replication when the master has initial data: https://dev.mysql.com/doc/refman/5.5/en/replication-howto-existingdata.html. It requires a dump and import. BTW this is not very different than MongoDB.

I thought we discussed this change and you indicated the only benefit from moving to a PV instead of an emptydir was removing the need to re-do a full replication on startup.  If that's not accurate, then I agree that there may not be value in describing how to do this with an emptydir.  In any case if my interpretation of your link is correct, then it means the existing templates never worked if a slave restarted (unless the db is smaller than the binlog, perhaps).  That could be fixed by updating the slave scripts to perform a remote dump of the master, on startup if the db directory is empty.

> And it is weird to have to justify why we use a PVC for the slave, while no mention as to why there is a PVC for master. IMHO it is not the case that one is more obvious than the other.

It seems obvious to me.  The master needs a PV because it is the store of record, and the slave does not need a PV because it is just a replica which bootstraps from the master when it comes up.  If that assumption is incorrect (that the slave can bootstrap from the master w/o additional intervention) then the original implementation was broken from the start, but could be (should be) fixed by performing the db dump step. It doesn't have to be fixed in this PR, but fixing it would make the mysql replica topology more useful since you could run N slaves for better read performance without needing N persistent volumes.  The other advantage to running slaves on an emptydir instead of a PV is that you can run replicas with a single replication controller, which you can't (currently) do with a PV.